app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -359,8 +368,62 @@ private int initializeAppResources() {
         }
         return STATE_UNINSTALLED;
     }
+    
+	public SqlStorage<ApplicationRecord> getInstalledAppRecords() {
+		return getGlobalStorage(ApplicationRecord.class);
+	}
+	
+	/** Return all ApplicationRecords that are installed and NOT archived **/
+	public ArrayList<ApplicationRecord> getVisibleAppRecords() {
+	    ArrayList<ApplicationRecord> visible = new ArrayList<ApplicationRecord>();
+        for (ApplicationRecord r : getInstalledAppRecords()) {
+            if (!r.isArchived()) {
+                visible.add(r);
+            }
+        }
+        System.out.println("NUM VISIBLE APPS FOUND: " + visible.size());
+        return visible;
+	}
+	
+	/**Return all ApplicationRecords that are installed AND are not archived
+	 * AND have MM verified **/
+	public ArrayList<ApplicationRecord> getReadyAppRecords() {
+		ArrayList<ApplicationRecord> ready = new ArrayList<ApplicationRecord>();
+		for (ApplicationRecord r : getVisibleAppRecords()) {
+			if (r.resourcesValidated()) {
+				ready.add(r);
+			}
+		}
+        System.out.println("NUM READY APPS FOUND: " + ready.size());
+		return ready;
+	}
+	
+	public boolean visibleAppsPresent() {
+	    boolean b = getVisibleAppRecords().size() > 0;
+	    return b;
+	}
+	
+    public ApplicationRecord[] appRecordArray() {
+        SqlStorage<ApplicationRecord> appList = CommCareApplication._().getInstalledAppRecords();

```

Comment:
Isn't the rest of this function just doing appList.toArray(new ApplicationRecord[0])?